crml/staking/src/tests.rs

```diff
@@ -267,26 +252,31 @@ fn multi_era_reward_should_work() {
 		assert_ok!(Staking::set_payee(Origin::signed(10), RewardDestination::Controller));
 
 		// Compute now as other parameter won't change
-		let total_payout_0 = current_total_payout::<Balances>();
+		Rewards::new_fiscal_era();
+		let total_payout_0: Balance = Rewards::calculate_next_reward_payout();
 		assert!(total_payout_0 > 10); // Test is meaningful if reward something
-		<Module<Test>>::reward_by_ids(vec![(11, 1)]);
+		Rewards::reward_by_ids(vec![(11, 1)]);
 
 		start_session(0);
 		start_session(1);
 		start_session(2);
 		start_session(3);
 
+		Staking::on_initialize(System::block_number() + 1);
+
 		assert_eq!(Staking::current_era(), 1);
 		assert_eq!(Balances::total_balance(&10), init_balance_10 + total_payout_0);
 
-		start_session(4);
+		start_session(5);
```

Comment:
This could stay as before but the next one which is triggering a new era should be start_session(6). The era length is 3, so sessions `0, 1, 2 => era 0` and sessions 3, 4, 5 => era 1, session 6 => era 2. This was working before because start_session was wrong.